### PR TITLE
DAOS-11517 test: deployment/network_failure.py - Create pool across 4…

### PR DESCRIPTION
**Pull request by shimizukko**
… ranks in 2 nodes with --nsvc=3

For the network isolation test, we used to create a pool
across two ranks in one node. However, this may cause an
issue with pool service replica when the network interface
is taken down.

Pool service leader can be located in any node regardless
of the pool’s location, so if we happen to take down the
interface on the node where the leader is located, the
pool service will not be accessible and will not be
recovered. Then the subsequent operations may fail.

To resolve this issue, create a pool across four ranks
in two nodes with `--nsvc=3`, 3 service replicas. Then we
should be able to recover even when the leader is
affected.

Also added the dmg system query check after bringing
down/up the network interface. When we update the
interface, some ranks may get excluded after a while.
i.e., they don't get excluded right after the update. Thus,
wait for 1 min, then repeatedly call dmg system query to
check the status.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: network_failure_isolation
Required-githooks: true

Signed-off-by: Makito Kano <makito.kano@intel.com>

### CODE REVIEW

**Comment by shimizukko** [comment]
I might have to resolve merge conflict when https://github.com/daos-stack/daos/pull/10087 merges.

File: src/tests/ftest/deployment/network_failure.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import os
 import time
 from ClusterShell.NodeSet import NodeSet
+from collections import defaultdict
```
Comment (daltonbohning) [inline src/tests/ftest/deployment/network_failure.py:9]: We should resolve this pylint error by moving the import
```
import time
from collections import defaultdict
from ClusterShell.NodeSet import NodeSet
```

File: src/tests/ftest/deployment/network_failure.py [hunk 2]
```diff
@@ -112,7 +113,37 @@ def create_ip_to_host(self):
...
+                ["response"]["members"].
+
+        Returns:
+            dict: Hostname to ranks mapping. Hostname (key) is string. Ranks (value) is
+                list of int.
+
+        """
+        host_to_ranks = defaultdict(list)
+        for member in system_query_members:
+            ip_addr = member["addr"].split(":")[0]
```
Comment (daltonbohning) [inline src/tests/ftest/deployment/network_failure.py:141]: Wouldn't `member` be the keys in the dictionary, not the values? Or maybe it's not a dictionary?

File: src/tests/ftest/deployment/network_failure.py [hunk 3]
```diff
@@ -303,43 +334,36 @@ def test_network_failure_isolation(self):
...
         self.log.info("network_down_host = %s", self.network_down_host)
 
-        # 2. Create a pool across two ranks on the same node; 0 and rank_r. We have to
-        # provide the size because we're using --ranks.
+        # 2. Create a pool across the four ranks on the two nodes. Use --nsvc=3. We have
+        # to provide the size because we're using --ranks.
         self.add_pool(namespace="/run/pool_size_value/*", create=False)
-        self.pool.target_list.update([0, rank_r])
+        self.pool.target_list.update(target_list)
         self.pool.create()
```
Comment (daltonbohning) [inline src/tests/ftest/deployment/network_failure.py:367]: FYI - this should work
```
        self.add_pool(namespace="/run/pool_size_value/*", target_list=target_list)
```

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/tests/ftest/deployment/network_failure.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/deployment/network_failure.py:8]: Done

File: src/tests/ftest/deployment/network_failure.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/deployment/network_failure.py:141]: dmg system query returns the following output.

```
{
  "response": {
    "members": [
      {
        "addr": "10.8.1.110:10001",
        "state": "joined",
        "fault_domain": "/wolf-110.wolf.hpdd.intel.com",
        "rank": 0,
        "incarnation": 851553937757569024,
        "uuid": "45b814df-7994-4be2-b8d6-89e106885ce2",
        "fabric_uri": "ofi+verbs;ofi_rxm://192.168.100.110:31317",
        "fabric_contexts": 19,
        "info": "",
        "last_update": "2022-09-08T23:55:24.330470073Z"
      },
```

`system_query_members` is the list of dictionaries in the `members` list.

File: src/tests/ftest/deployment/network_failure.py (hunk 3 above)
Comment (shimizukko) [reply src/tests/ftest/deployment/network_failure.py]: Done. Thanks.

File: src/tests/ftest/deployment/network_failure.py [hunk 4]
```diff
@@ -362,8 +379,8 @@ def test_network_failure_isolation(self):
 
         # 2. Create a pool across the four ranks on the two nodes. Use --nsvc=3. We have
         # to provide the size because we're using --ranks.
-        self.add_pool(namespace="/run/pool_size_value/*", create=False)
-        self.pool.target_list.update(target_list)
+        self.add_pool(
+            namespace="/run/pool_size_value/*", create=False, target_list=target_list)
         self.pool.create()
```
Comment (daltonbohning) [inline src/tests/ftest/deployment/network_failure.py:384]: The `self.pool.create()` isn't necessary, since `create=True` is the default

Comment (shimizukko) [reply src/tests/ftest/deployment/network_failure.py]: Done. Thanks for pointing out.

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

**Pull request merged**